### ISSUE TITLE
Reload manifest without page refresh

### ISSUE THREAD

**Issue by lavigne** [ProjectMirador/mirador#779]
I'm working with the downloadable release version of Mirador.  When I load the same manifest a second time without refreshing the page, I get a Javascript error on line 19580 of the embedded openseadragon.  The error is "Uncaught TypeError: Cannot read property 'layoutAddress' of undefined".  This seems to be due to the fact that Mirador doesn't request the thumbnails the second time I load the manifest, I can see this because I'm running an http debugging proxy and I see the request for the manifest return just fine but the subsequent image requests are not sent like they are on the original load of the manifest.  I built Mirador from source and the problem seems to be fixed.  However, I'm getting several 404s on things Mirador (actually the embedded openseadragon) is looking for in the "build" directory.  For example: /build/mirador/locales/dev/translation.json.  Some of these files exist, such as /build/mirador/locales/en/translation.json, but other such as /mirador/locales/dev/translation.json do not. Either way, it seems incorrect that we would be requesting these files from the "build" directory, I would expect a relative path such as locales/en/translation.json.  I don't really want to put a build directory in my project just to satisfy these broken links but I don't want to modify the Mirador source to fix the issue either.  Perhaps there is an updated version of OpenSeadragon that we could embed?  Also, is there going to be another release any time in the near future?  I'm not really comfortable going to production with a source build.  Here is the code I'm using to initialize the viewer for your reference:

``` javascript
Mirador({
    "id" : "viewer",
    "layout" : "1x1",
    "saveSession" : false,
    "data" : [{
        "manifestUri" : manifestUrl
    }],
    "mainMenuSettings" : {
        "show" : false
    },        
    "windowObjects" : [{
        "loadedManifest" : manifestUrl,
        "viewType" : "ImageView"
    }]
});
```

**Comment by aeschylus** [comment]
@lavigne, thanks for reporting this. It seems like two separate issues, but both have to do with refining the developer API so Mirador can be better integrated in larger systems.

I'm not sure I understand the thumbnail problem, can you describe that in a another way?

The relative paths issue is definitely a needed upgrade now that users are demanding to integrate Mirador into larger applications instead of simply putting it on webpages. We should get to that.

There is a new release (2.1) coming in the next Month or so. A major priority of that release is to update the development process to include much more frequent releases (in a kind of "continuous deployment" or continuous release).

**Comment by lavigne** [comment]
@aeschylus , Thanks for the quick response.  About a minute after I posted this issue I found the "buildPath" configuration in the Mirador source.  As a workaround I'm able to fix the 404s by changing that path to point to the location of mirador in my project and adding a couple of missing directories and their corresponding translation.json.  For your records, the missing files are /locales/dev/translation.json and /locales/en-US/translation.json.  I just created the missing directories and put the en/translation.json file in them.  A hack, but it works until a fix can be put in.

The thumbnail issue requires a little more explanation.  Our app has a search function that allows users to search for manuscripts and printed books.  They can click on search result and the Mirador viewer is loaded for the search result using the Mirador code I posted above.  Basically, we hide the search results list, show the "viewer" div and call the code I posted above to load the images into Mirador.  If the user searches again and clicks on the same search result, the viewer doesn't load the images.  Using an http debugger, I can see that on the first call to the Mirador code, Mirador  fetches the manifest, info.json and a bunch of images.  The second time I call the code with the same manifest, only the call to the manifest goes up.  If I load a different manifest, it works just fine.  However, once I've loaded a manifest into the viewer, I'm never able to load it again until I refresh the page.  Does that make sense?  Also, as I mentioned before, the issue is fixed in the latest build.
